common/core/web/tools/sentry-manager/src/index.ts

```diff
@@ -0,0 +1,137 @@
+///<reference path="../node_modules/@sentry/browser/build/bundle.min.js" />
+
+namespace com.keyman {
+  /**
+   * Controls whether or not the generated Sentry event is logged to the console (true)
+   * or sent to our Sentry server (false).
+   */
+  let DEBUG = false;
+
+  type Options = typeof KeymanSentryManager.DEFAULT_OPTIONS;
+
+  export class KeymanSentryManager {
+    keymanPlatform: string;
+
+    static STANDARD_ALIASABLE_FILES = {
+      'keymanweb.js':    'keymanweb.js',
+      'kmwuibutton.js':  'kmwuibutton.js',
+      'kmwuifloat.js':   'kmwuifloat.js',
+      'kmwuitoggle.js':  'kmwuitoggle.js',
+      'kmwuitoolbar.js': 'kmwuitoolbar.js'
+      // Also add entries for the naming system used by Android and iOS - and map them to the EMBEDDED upload, not the std 'native' one.
+    }
+
+    static DEFAULT_OPTIONS = {
+      hostPlatform: "native-web"
+    }
+
+    constructor(options: Options = KeymanSentryManager.DEFAULT_OPTIONS) {
+      this.keymanPlatform = options.hostPlatform;
+    }
+
+    // If we've recognized one of our source files, 
+    aliasFilename(filename: string): string|null {
+      if(!this.mayAlias(filename)) {
+        return null;
+      }
+
+      filename = KeymanSentryManager.STANDARD_ALIASABLE_FILES[filename];
+
+      switch(location.protocol) {
+        case 'http:':
+          return 'http://' + location.host + '/' + filename;
+        case 'file:':
+          return 'file:///' + filename;
+        default:
+          return null;
+      }
+    }
+
+    mayAlias(filename: string): boolean {
+      return !!KeymanSentryManager.STANDARD_ALIASABLE_FILES[filename];
+    }
+
+    // Filters all expected but unnecessary path prefixes, affixes, and suffixes reported by Sentry from our products.
+    // This allows us to mask all different sorts of installations with a single uploaded path.
+    pathFilter(event: any) {
+      // Get the underlying JS error.
+      let exception = event.exception;
+
+      // Iterate through all wrapped exceptions.
+      for(let e of exception.values) {
+        for(let frame of e.stacktrace.frames) {
+          let URL = frame.filename as string;
+          let filename: string = '';
+          try {
+            filename = URL.substr(URL.lastIndexOf('/')+1);
+            if(this.mayAlias(filename)) {
+              // Make the file look like it's at the host's root so that the Sentry server will recognize it
```

Comment:
I wonder if we use `/keymanweb/`... rather than the root or some other unique name that's unlikely to be confused in the future?

Reply:
Please note https://docs.sentry.io/platforms/javascript/sourcemaps/#uploading-source-maps-to-sentry and compare against [the new line 408 of build.sh](https://github.com/keymanapp/keyman/pull/3122/files#diff-7523263e3a0414ebc35bbfa2c366863bR408).

Short version:  the `sentry-cli` tool for uploading uses the local file structure to determine what path the Sentry server will look for when attempting to match error logs against the source.  We can 'strip' a prefix from that path easily enough, but inserting custom path elements will require file-folder manipulation during builds, so far as I can tell.

Reply:
OK. No problemo 👍